src/sensors.c

```diff
@@ -88,6 +88,17 @@ static int detect_sony_sensor(sensor_ctx_t *ctx, int fd,
     if (i2c_change_addr(fd, i2c_addr) < 0)
         return false;
 
+    int chip_id = READ(0x57);
+    if(chip_id == 0x06) {
+        sprintf(ctx->sensor_id, "IMX347");
+        return true;
+    }
+#if 0
```

Comment:
I tested both ifdef part on IMX335, and it seems to work. I suggest removing ifdef from the code and stick with one IMX335 detection method (because in a year it will be near impossible to understand why they are both exist)

We can have "backup" method in text comment with detailed explanation in case we need to distinguish more sensors in the future